### ISSUE TITLE
Unable to apply migration #11

### ISSUE THREAD

**Issue by seth586** [lightningnetwork/lnd#3606]
### Background

First run of 0.8.0 from 0.7.1 results in the following console output:

```
[INF] CHDB: Unable to apply migration #11
[ERR] LTND: Unable to open channeldb: cannot upgrade with invoice(s) in accepted state, see release notes
```

Release notes state `Note that you cannot update if you have any pending hodl invoice.`, however `lncli` documentation is unclear how to identify and remove these invoices.

### Your environment

* Migrating `lnd` 0.7.1 to 0.8.0
* which operating system (`FreeBSD 11.2`)
* `bitcoind 0.18.1`

**Comment by seth586** [comment]
Solution:

Hold invoices were created by [sparkswap](https://github.com/sparkswap), identified in [RTL](https://github.com/ShahanaFarooqui/RTL) as 'paid' but not 'settled'.

`lncli cancelinvoice` was used to cancel these invoices, and the database migrated without issue.

**Comment by dannypaz** [comment]
@seth586 can you provide more info on your solution? Did you need to downgrade your version of LND?

**Comment by dannypaz** [comment]
So context, the update to 0.8.0 failed for us because we had HODL invoices in the ACCEPTED state... and migration 11 would error out on LND because of it

Our LND node was unable to start because of this bad state giving us no way to change the state of the invoices (womp) so we created a docker image of an older version of LND, cherrypicked the neutrino mainnet commit and were able to load our old files from the node.

We used the CLI to grab all the ACCEPTED r_hashes from channel db, converted the r_hashes from base64 to hex and were able to cancel all of them.

We then copied channel.db back into the zap directory and everything worked.

**Comment by Overtorment** [comment]
If anyone is struggling:
Do `lncli listinvoices  --max_invoices 100500 | grep state | sort -u` and if there is anything in `ACCEPTED` state then you need to find this invoice, get its payment hash and run `lncli cancelinvoice PAYMENT_HASH`. Hash has to be in hex format, not base64.
Repeat until there are no ACCEPTED invoices left.

**Comment by Technifocal** [comment]
I feel really stupid, but I'm on version `0.7.1-beta` (trying to upgrade after not using LND for a while) and I don't have `cancelinvoice`:

```
$ lncli cancelinvoice
No help topic for 'cancelinvoice'
```

`--help` also doesn't list it:

```
Payments:
     sendpayment    Send a payment over lightning.
     payinvoice     Pay an invoice over lightning.
     sendtoroute    Send a payment over a predefined route.
     addinvoice     Add a new invoice.
     lookupinvoice  Lookup an existing invoice by its payment hash.
     listinvoices   List all invoices currently stored within the database. Any active debug invoices are ignored.
     listpayments   List all outgoing payments.
     queryroutes    Query a route to a destination.
     decodepayreq   Decode a payment request.
     fwdinghistory  Query the history of all forwarded HTLCs.
```

How would I go about cancelling the invoices?

**Comment by wpaulino** [comment]
@Technifocal the command is available when compiling `lnd` with `make install tags=invoicesrpc`.

**Comment by Technifocal** [comment]
I was able to fix it by downloading `v0.8.0-beta-rc1`, which both allowed me to boot and had `cancelinvoice`.

https://github.com/lightningnetwork/lnd/releases/tag/v0.8.0-beta-rc1

Here was my script:

`lncli listinvoices --max_invoices 9999999 | jq -r '.invoices[] | select(.state=="ACCEPTED").r_hash' | while read invoiceHash; do lncli cancelinvoice "$(base64 -d <<< "${invoiceHash}" | xxd -pr | tr -d '\n')"; done`

@wpaulino Thank you.
